### DISCUSSION TITLE
[Feature Request]Something similar to `tokio::runtime::Builder::on_thread_start()` for actix-web workers

### DISCUSSION THREAD

**Discussion by SteveLauC** [actix/actix-web#3366] [Ideas] [answered]
Hi, thanks for this amazing project!

I would like to know do we have something similar to [`tokio::runtime::Builder::on_thread_start()`](https://docs.rs/tokio/latest/tokio/runtime/struct.Builder.html#method.on_thread_start) for workers, so that one can do something when those workers start, e.g., set CPU affinity.

**Comment by SteveLauC** [comment]
I implemented this in my [fork](https://github.com/SteveLauC/actix-web/tree/feat/on_worker_start), though I still hope we can have this in the official release:>

**Comment by robjtede** [comment, marked as answer]
I added [an example of how to achieve CPU core pinning](https://github.com/actix/actix-web/blob/master/actix-web/examples/worker-cpu-pin.rs).

I'd prefer not to add an `on_thread_start` method to `HttpServer` since the app factory functions are run per-worker already.
